src/lib/Form/Factory/FormFactory.php

```diff
@@ -1258,4 +1276,39 @@ public function editUser(
 
         return $this->formFactory->createNamed($name, UserEditType::class, $data, $options);
     }
+
+    /**
+     * @param \EzSystems\EzPlatformAdminUi\Form\Data\Location\LocationTrashWithAssetData|null $data
+     * @param string|null $name
+     *
+     * @return \Symfony\Component\Form\FormInterface
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    public function trashLocationWithAsset(
+        LocationTrashWithAssetData $data = null,
+        ?string $name = null
+    ): FormInterface {
+        $name = $name ?: StringUtil::fqcnToBlockPrefix(LocationTrashWithAssetType::class);
+        $location = $data->getLocation();
+        $options = [];
+        if (null !== $location) {
+            $contentName = $location->getContent()->getName();
+            $contentType = $this->contentTypeService
+                ->loadContentType(
+                    $location->getContentInfo()->contentTypeId)
+                ->getName(
+                    $location->getContentInfo()->mainLanguageCode
+            );
+            $translatorParameters = ['%content_name%' => $contentName, '%content_type%' => $contentType];
+            $options['trashAssetsOptions'] = [
```

Comment:
Should we build those options here or transfer them from the controller? ping @webhdx

Reply:
IMHO this should logic should be placed in `EzSystems\EzPlatformAdminUi\Form\Type\Location\LocationTrashWithAssetType`. For instance using form events: https://symfony.com/doc/current/form/dynamic_form_modification.html#form-events-user-data